presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/TypeValidator.java

```diff
@@ -154,10 +155,11 @@ private void checkFunctionCall(Map<Symbol, FunctionCall> functionCalls)
             }
         }
 
-        private static void verifyTypeSignature(Symbol symbol, TypeSignature expected, TypeSignature actual)
+        private void verifyTypeSignature(Symbol symbol, TypeSignature expected, TypeSignature actual)
         {
             // UNKNOWN should be considered as a wildcard type, which matches all the other types
-            if (!actual.equals(UNKNOWN.getTypeSignature())) {
+            TypeManager typeManager = metadata.getTypeManager();
+            if (!actual.equals(UNKNOWN.getTypeSignature()) && !typeManager.isTypeOnlyCoercion(typeManager.getType(actual), typeManager.getType(expected))) {
```

Comment:
@martint why do we need the `isTypeOnlyCoercion` functionality, is it just to avoid the cost of calling an identity `MethodHandle`? I find it very confusing that we don't include these in the plan